storage/src/key_value/mod.rs

```diff
@@ -33,18 +33,20 @@ use agent::Agent;
 pub type Value<'a> = Cow<'a, [u8]>;
 
 pub trait KeyValueStorage {
-    fn get<'a>(&'a mut self, column: KeyValueColumn, key: &[u8]) -> Result<Option<Value<'a>>>;
+    fn get<'a>(&'a self, column: KeyValueColumn, key: &[u8]) -> Result<Option<Value<'a>>>;
 
-    fn exists(&mut self, column: KeyValueColumn, key: &[u8]) -> Result<bool>;
+    fn exists(&self, column: KeyValueColumn, key: &[u8]) -> Result<bool>;
 
-    fn get_column_keys<'a>(&'a mut self, column: KeyValueColumn) -> Result<Vec<Value<'a>>>;
+    fn get_column_keys<'a>(&'a self, column: KeyValueColumn) -> Result<Vec<Value<'a>>>;
 
-    fn get_column<'a>(&'a mut self, column: KeyValueColumn) -> Result<Vec<(Value<'a>, Value<'a>)>>;
```

Comment:
these must be `mut` to ensure proper thread safety. please do not change the mutability of these APIs. rocksdb crate doesn't ensure thread safety. Feel free to call rocksDB directly if you need it.